crates/docker-utils/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+use std::{
+    process::{Command, Stdio},
+    time::Duration,
+};
+
+use anyhow::Context;
+
+pub struct DockerContainerGuard {
+    container_id:   String,
+    container_port: u16,
+}
+
+impl DockerContainerGuard {
+    pub fn port(&self) -> u16 {
+        self.container_port
+    }
+}
+
+impl Drop for DockerContainerGuard {
+    fn drop(&mut self) {
+        if let Err(err) = run_cmd(&format!("docker stop {}", &self.container_id)) {
+            eprintln!("Failed to stop docker container: {}", err);
+        }
+
+        // Redundant, but better safe than sorry
+        if let Err(err) = run_cmd(&format!("docker rm {}", &self.container_id)) {
+            eprintln!("Failed to remove docker container: {}", err);
+        }
+    }
+}
+
+/// Starts a postgres docker container that will accept all connections with a
+/// random port assigned by the os or docker. The container will be stopped and
+/// removed when the guard is dropped.
+///
+/// Note that we're using sync code here so we'll block the executor - but this
+/// is fine, because the spawned container will still run in the background.
+pub async fn setup() -> anyhow::Result<DockerContainerGuard> {
```

Comment:
Given the crate is called `docker-utils`, this name isn't descriptive enough. Either the crate should be something like `pg-docker-utils`, or the function name should be more descriptive.

Reply:
Renamed to `postgres-docker-utils`